### PR TITLE
feature/leave-spaces-in-docblock

### PR DESCRIPTION
**Pull request by alexndlm**
-leave spaces in doc comments
-update packages
-refactoring

### CODE REVIEW

File: src/Rector/AddSeeAnnotationRector.php [hunk 1]
```diff
@@ -92,62 +93,62 @@ public function refactor(Node $node): ?Node
...
         }
     }
 
     /**
      * Checks test method.
      */
-    private function checkTestMethod(Class_ $class, ClassMethod $classMethod): void
+    private function checkTestMethod(Class_ $classNode, ClassMethod $classMethod): void
     {
         /** @var \Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo|null $phpDocInfo */
```
Comment (itorgov) [inline src/Rector/AddSeeAnnotationRector.php:141]: Please remove `null`.

File: src/Rector/PhpDocBlockTrait.php [hunk 2]
```diff
@@ -21,77 +21,58 @@ trait PhpDocBlockTrait
...
-                        }
                         $commentText = \preg_replace(['/^\/\/\s*/', '/^\s*\*?\s*/'], '', $commentText);
-                        $docCommentText .= "\n *" . ($commentText ? ' ' . $commentText : '');
+                        $newComments[] = new PhpDocTextNode($commentText);
                     }
                 }
             }
+
+            $newComments[] = new PhpDocTextNode('');
+            $phpDocInfo->getPhpDocNode()->children = array_merge($newComments, $phpDocInfo->getPhpDocNode()->children);
```
Comment (itorgov) [inline src/Rector/PhpDocBlockTrait.php:57]: `\array_merge`

File: src/Rector/PhpDocCommentRector.php [hunk 3]
```diff
@@ -266,7 +266,8 @@ private function checkTextNode(PhpDocTextNode $phpDocTextNode): void
 
         $newText = \implode(\PHP_EOL, $text);
 
-        if ($newText !== $phpDocTextNode->getAttribute('orig_node')->text) {
+        if ($phpDocTextNode->getAttribute('orig_node')
```
Comment (itorgov) [inline src/Rector/PhpDocCommentRector.php:269]: `$phpDocTextNode->getAttribute('orig_node') !== null`

File: tests/Rector/PhpDocReturnForIterableRector/Fixture/parent_method_has_docblock.php.inc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace EonX\EasyQuality\Tests\Rector\PhpDocReturnForIterableRector\Fixture;
+
+use EonX\EasyQuality\Tests\Rector\PhpDocReturnForIterableRector\Source\ParentTestCase;
+
+abstract class AbstractParentTestCase{
+    /**
+     * @return iterable<mixed>
+     */
+    abstract public static function provideData1(): iterable;
```
Comment (itorgov) [inline tests/Rector/PhpDocReturnForIterableRector/Fixture/parent_method_has_docblock.php.inc:12]: Why do you use `1` at the end?

File: tests/Rector/ReturnArrayToYieldRector/Fixture/should_not_remove_doc_blocks.php.inc [hunk 5]
```diff
@@ -9,9 +9,9 @@ final class ShouldNotRemoveComment extends ParentTestCase
     public function provideData(): array
     {
         /**
-         * @doto Let's pretend that this comment is extremely important and meaningful.
-         *
          * It should not be removed by Rector.
+         *
+         * @doto Let's pretend that this comment is extremely important and meaningful.
```
Comment (itorgov) [inline tests/Rector/ReturnArrayToYieldRector/Fixture/should_not_remove_doc_blocks.php.inc:14]: Maybe `@todo`?

**Review by itorgov — CHANGES_REQUESTED** [review]

File: src/Rector/AddSeeAnnotationRector.php (hunk 1 above)
Comment (alexndlm) [reply src/Rector/AddSeeAnnotationRector.php]: Done.

File: src/Rector/PhpDocBlockTrait.php (hunk 2 above)
Comment (alexndlm) [reply src/Rector/PhpDocBlockTrait.php]: Done.

File: src/Rector/PhpDocCommentRector.php (hunk 3 above)
Comment (alexndlm) [reply src/Rector/PhpDocCommentRector.php]: Done.

File: tests/Rector/PhpDocReturnForIterableRector/Fixture/parent_method_has_docblock.php.inc (hunk 4 above)
Comment (alexndlm) [reply tests/Rector/PhpDocReturnForIterableRector/Fixture/parent_method_has_docblock.php.inc:12]: Done.

File: tests/Rector/ReturnArrayToYieldRector/Fixture/should_not_remove_doc_blocks.php.inc (hunk 5 above)
Comment (alexndlm) [reply tests/Rector/ReturnArrayToYieldRector/Fixture/should_not_remove_doc_blocks.php.inc]: Done.

File: src/Rector/PhpDocReturnForIterableRector.php [hunk 6]
```diff
@@ -95,21 +98,55 @@ public static function getSubscribedEvents(): iterable
...
         }
 
         return $hasChanged ? $classMethod : null;
     }
+
+    private function isParentMethodHasDocBlock(Node $classMethod): bool
+    {
+        $scope = $classMethod->getAttribute(AttributeKey::SCOPE);
+        if (!$scope instanceof Scope) {
+            // possibly trait
```
Comment (roman-eonx) [inline src/Rector/PhpDocReturnForIterableRector.php:126]:
```suggestion
            // Possibly a trait
```

File: src/Rector/PhpDocReturnForIterableRector.php [hunk 7]
```diff
@@ -95,21 +98,55 @@ public static function getSubscribedEvents(): iterable
...
+        if (!$classReflection instanceof ClassReflection) {
+            return false;
+        }
+
+        /** @var string $methodName */
+        $methodName = $this->getName($classMethod->name);
+
+        foreach ($classReflection->getParents() as $parentClassReflection) {
+            $nativeClassReflection = $parentClassReflection->getNativeReflection();
+            // the class reflection aboves takes also @method annotations into an account
```
Comment (roman-eonx) [inline src/Rector/PhpDocReturnForIterableRector.php:140]:
```suggestion
            // The class reflection above takes also @method annotations into an account
```

File: tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc [hunk 8]
```diff
@@ -7,14 +7,14 @@ use EonX\EasyQuality\Tests\Rector\PhpDocReturnForIterableRector\Source\ParentTes
...
+     *
      * @todo Some sing.
      * @cutomTag Tag value.
      *
-     * Custom Text
-     *
      * @return iterable
      *
-     * Another Custom Text
+     * Another Custom Text that appear in the bottom
```
Comment (roman-eonx) [inline tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc:17]:
```suggestion
     * Another Custom Text that appears at the bottom
```

File: tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc [hunk 9]
```diff
@@ -33,14 +33,14 @@ use EonX\EasyQuality\Tests\Rector\PhpDocReturnForIterableRector\Source\ParentTes
...
+     *
      * @todo Some sing.
      * @cutomTag Tag value.
      *
-     * Custom Text
-     *
      * @return iterable<mixed>
      *
-     * Another Custom Text
+     * Another Custom Text that appear in the bottom
```
Comment (roman-eonx) [inline tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc:43]:
```suggestion
     * Another Custom Text that appears at the bottom
```

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc (hunk 9 above)
Comment (alexndlm) [reply tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc]: Done.

File: tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc (hunk 8 above)
Comment (alexndlm) [reply tests/Rector/PhpDocReturnForIterableRector/Fixture/fixture3.php.inc]: Done

File: src/Rector/PhpDocReturnForIterableRector.php (hunk 7 above)
Comment (alexndlm) [reply src/Rector/PhpDocReturnForIterableRector.php]: Done

File: src/Rector/PhpDocReturnForIterableRector.php (hunk 6 above)
Comment (alexndlm) [reply src/Rector/PhpDocReturnForIterableRector.php]: Done

**Review by roman-eonx — APPROVED** [review]

**Pull request merged**